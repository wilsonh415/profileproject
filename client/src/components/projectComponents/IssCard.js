import React from 'react';
import { Card, Nav, Button } from 'react-bootstrap';
import issfoto from '../../images/issfoto.jpeg';


class IssCard extends React.Component {
    render() {
        const issCardStyle = {
            backgroundColor: "white",
            border: "1px solid",
            borderRadius: "5px",
            minWidth: "286px",
            maxWidth: "286px",
            marginBottom: "5vh",
            minHeight: "477px",
            maxHeight: "477px",
            position: "relative"
        };
        return (
            <div style={issCardStyle}>
                <Card>
                    <Card.Img variant="top" src={issfoto} width={286} height={180}/>
                    <Card.Body>
                        <Card.Title><b>Where is the International Space Station (ISS)?</b></Card.Title>
                        <b><hr/></b>
                        <Card.Text>
                        Launched in November 20, 1998, the ISS orbits the Earth every 93 minutes.
                        </Card.Text>
                        <b><hr/></b>
                        <Nav.Link href="/Iss">
                            <Button variant="primary">Locate ISS!</Button>
                        </Nav.Link>
                    </Card.Body>
                </Card>
            </div>
        )
    }
}

export default IssCard;