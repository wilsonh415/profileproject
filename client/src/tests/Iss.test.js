import React from 'react';
import IssCard from '../components/projectComponents/IssCard';
import renderer from 'react-test-renderer';

// snapshot test
test('Page changes when "Locate ISS!" is clicked', () => {
    const component = renderer.create(
        <IssCard></IssCard>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});