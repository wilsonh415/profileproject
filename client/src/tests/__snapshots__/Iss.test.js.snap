// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page changes when "Locate ISS!" is clicked 1`] = `
<div
  style={
    Object {
      "backgroundColor": "white",
      "border": "1px solid",
      "borderRadius": "5px",
      "marginBottom": "5vh",
      "maxHeight": "477px",
      "maxWidth": "286px",
      "minHeight": "477px",
      "minWidth": "286px",
      "position": "relative",
    }
  }
>
  <div
    className="card"
  >
    <img
      className="card-img-top"
      height={180}
      src="issfoto.jpeg"
      width={286}
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        <b>
          Where is the International Space Station (ISS)?
        </b>
      </div>
      <b>
        <hr />
      </b>
      <p
        className="card-text"
      >
        Launched in November 20, 1998, the ISS orbits the Earth every 93 minutes.
      </p>
      <b>
        <hr />
      </b>
      <a
        className="nav-link"
        href="/Iss"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        <button
          className="btn btn-primary"
          disabled={false}
          type="button"
        >
          Locate ISS!
        </button>
      </a>
    </div>
  </div>
</div>
`;
